launchflow/workflows/destroy_resource_tofu/delete_tofu_resource.py

```diff
@@ -1,22 +1,56 @@
 from launchflow import exceptions
 from launchflow.gcp_clients import get_storage_client
+from launchflow.models.enums import ResourceProduct
 from launchflow.workflows.commands.tf_commands import TFDestroyCommand
 from launchflow.workflows.destroy_resource_tofu.schemas import DestroyResourceTofuInputs
 from launchflow.workflows.utils import run_tofu
 
 
 async def delete_tofu_resource(inputs: DestroyResourceTofuInputs):
+    # TODO: there's a bug in here we need to fix before rolling out this
+    # If the initial create fails the inputs aren't recorded and the user is unable
+    # to destroy the resource. This could happen if the user isn't logged in or something.
+    # I think the fix is to just make sure the inputs are always recorded.
     state_prefix = inputs.launchflow_uri.tf_state_prefix(module=inputs.resource.product)
+    tf_vars = inputs.resource.inputs or {}
+    module_dir = inputs.resource.product
+    module_dir = f"resources/{inputs.resource.product}"
+    if inputs.resource.product == "unknown":
+        return
+    # TODO: I don't like this hard coding but it works for now
+    is_k8_resource = (
+        inputs.resource.product == ResourceProduct.KUBERNETES_SERVICE_CONTAINER
```

Comment:
should this be production == PRODUCT.value?

Reply:
done